CONTRIBUTING.md

```diff
@@ -0,0 +1,33 @@
+# Contributing
+
+Both the [dokku ansible role](https://github.com/dokku/ansible-dokku) and [dokku](https://github.com/dokku/dokku) itself are open source projects and contributions are highly welcome!
+
+## Codebase overview
+
+ * The role's directory layout follows [standard ansible practises](https://galaxy.ansible.com/docs/contributing/creating_role.html#roles).
+ * Besides the yaml-based ansible instructions, the role includes several new ansible *modules* in the `library/` folder (e.g. `dokku_app`).
+ * The `README.md` of this repository is auto-generated: do *not* edit it directly.  
+   In order to update it, run `make generate`.
+
+## Setting up a test environment
+
+This role is tested using [molecule](https://molecule.readthedocs.io/en/latest/).
+Setting up a test environment involves the following steps:
+
+ * Install [docker](https://www.docker.com/)
+ * Install [python](https://www.python.org/)
+ * Run `pip install -r  requirements.txt`
```

Comment:
Should we run tests in a virtualenv?

Reply:
I certainly use virtual environments for this; happy to mention it as an optional step.
One can also switch to something like tox but in my view it also makes things more complicated to understand.

Reply:
Hmmm up to you. I basically want to make sure we can cleanly run things without ruining someone's laptop :)